pandas/core/generic.py

```diff
@@ -5201,60 +5201,209 @@ def abs(self):
         """
         return np.abs(self)
 
-    _shared_docs['describe'] = """
-        Generate various summary statistics, excluding NaN values.
+    def describe(self, percentiles=None, include=None, exclude=None):
+        """
+        Generates descriptive statistics that summarize the central tendency,
+        dispersion and shape of a dataset's distribution, excluding
+        ``NaN`` values.
 
-        Parameters
-        ----------
-        percentiles : array-like, optional
-            The percentiles to include in the output. Should all
-            be in the interval [0, 1]. By default `percentiles` is
-            [.25, .5, .75], returning the 25th, 50th, and 75th percentiles.
-        include, exclude : list-like, 'all', or None (default)
-            Specify the form of the returned result. Either:
-
-            - None to both (default). The result will include only
-              numeric-typed columns or, if none are, only categorical columns.
-            - A list of dtypes or strings to be included/excluded.
-              To select all numeric types use numpy numpy.number. To select
-              categorical objects use type object. See also the select_dtypes
-              documentation. eg. df.describe(include=['O'])
-            - If include is the string 'all', the output column-set will
-              match the input one.
+        Analyzes both ``numeric`` and ``object`` series, as well
+        as `DataFrame` column sets of mixed data types.
 
-        Returns
-        -------
-        summary: %(klass)s of summary statistics
+        For ``numeric`` data, the result's index will include ``count``,
```

Comment:
I would leave these details in the notes section as it was before. Reason is that otherwise it 'takes a long time' before you get to the 'Parameters'. But maybe we could include in the previous sentence something like "Output format depends on the data type, see Notes for more details"